### ISSUE TITLE
Can't Build, missing build script

### ISSUE THREAD

**Issue by patrick-rodgers** [pnp/sp-js-provisioning#5]
#### Category
[ ] Enhancement
[X] Bug
[ ] Question

#### Expected / Desired Behavior / Question
Trying to execute a release following the instructions from #4.

#### Observed Behavior
```Console
λ npm run build
npm ERR! Windows_NT 10.0.17763
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\patrodg\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "run" "build"
npm ERR! node v8.11.3
npm ERR! npm  v3.10.8

npm ERR! missing script: build
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     C:\github\sp-js-provisioning\npm-debug.log
```

#### Steps to Reproduce
1. Clone Repo
2. npm install
3. `npm run build`

Also note there is no build script defined in the package.json.

**Comment by patrick-rodgers** [comment]
@olemp can you have a quick look please?

**Comment by olemp** [comment]
Oh, sorry, my bad @patrick-rodgers. It's `gulp build`. We could add an npm task for `build` aswell though.

**Comment by patrick-rodgers** [comment]
I am not really concerned with exactly the commands to run, just that they are documented and understood. I am assuming when you tested you were using gulp build?

**Comment by olemp** [comment]
Yes, I did, haven't really done any edits to the build setup. We could create an issue for improving the build if you want.

**Comment by patrick-rodgers** [comment]
Couple more questions:

- Running `gulp build` changes the files and git now sees uncommitted changes, is that expected? Do I ignore it? Do I need to commit those changes?
- It looks like you are checking in the built code under "lib" - should I just do a publish of what is there and skip the build, or should that be removed from github?

I don't mean to tell anyone what to do, this is a community project. My only concern is ensuring once published it works as you (and consumers) expect. I am not going to go through and test things - so relying on folks contributing to ensure things are correct.

Edit: I will add that since this was branched (years ago?) we have made significant changes to how the build works in pnpjs. Some of it might be helpful here and other parts likely not as this is still a single library repo. But there are likely improvements that could be made to the build process, if you wanted.

**Comment by olemp** [comment]
No problem @patrick-rodgers. While developing this "lib", I've pushed lib to git to so I could include the git reference in other projects using this lib. We should probably ignore lib in .gitignore, and make sure lib is included for npm publish. Sorry for forgetting this step!

**Comment by patrick-rodgers** [comment]
No worries, just want to have it be successful once published. Do you want to make a few changes and let me know and I'll re-pull and publish?

**Comment by olemp** [comment]
Will do @patrick-rodgers, thanks 👍

**Comment by olemp** [comment]
Fixed lib and build issue in https://github.com/pnp/sp-js-provisioning/pull/8.
